zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java

```diff
@@ -208,8 +209,13 @@ RuntimeException errorReading(String message, Throwable e) {
   RuntimeException errorStoringSpans(List<Span> spans, Throwable e) {
     metrics.incrementSpansDropped(spans.size());
     // The exception could be related to a span being huge. Instead of filling logs,
-    // print trace id, span id pairs
-    StringBuilder msg = appendSpanIds(spans, new StringBuilder("Cannot store spans "));
+    // print trace id/span id pairs, and a service name
+    StringBuilder messagePrefix = new StringBuilder("Cannot store spans ");
+    if (spans.size() > 0) {
+      messagePrefix.append("from ").append(spans.get(0).localServiceName()).append(' ');
```

Comment:
Quick question: are we comfortable with this change?  I missed a test that broke as a result of this.  My goal is to provide increased/easier visibility into which service is getting dropped.  IDs alone don't provide that level of transparency.

Reply:
A problem I see here is this may not be reliable. For instance there have been discussions of having a relay or proxy mode for the server that _could_ act as an aggregator of spans from some subset of services, and then forward them out using a zipkin reporter/sender. For instance if the relay was deployed one per host as a daemon exposing http (supported by all language libs) and forwarding over kafka or kinesis or something like that

Reply:
Would that make the service name here inaccurate?  If we get the name of the aggregator/proxy, then we at least still know the source of what tried to report the spans.  If we get the name of the service being proxied then that's even better imo.  But both seem acceptable to me.

Reply:
I didn't know we were logging this to begin with. Are the trace/span IDs logged even always accurate? At least with Elasticsearch, there can be a partial failure in the bulk request, and I don't think we can know which were successful and which failed. Probably our counter on dropped span count/bytes is off in case of partial failure also.
In setups I'm aware of today, at least the service name seems like it should be accurate.

Reply:
@shakuzen I didn't realize that there could be a partial failure in ES bulk requests. It seems like we have some issues here that probably need to be done in a separate PR as no-one will know to look here.

@Logic-32 I suggest dropping this commit possibly raising a new PR saying something like failure writing traces from the following (possibly truncated due to size of string) services

In that PR we can discuss the nuance of partial failure etc, which is possible anyway. This is debug so I suspect many won't keep this on always anyway. I think I wrote it. I'm not tagging him directly now but we would tag Chris Wensel also as they've had problems analyzing busted traces.. I'm not sure if this would help or not.

Reply:
I think I'll hold off on the new PR for now.  Seems like more hassle than it's worth.  Working on reverting now.